mlflow/pyspark/ml/__init__.py

```diff
@@ -925,8 +939,56 @@ def _log_posttraining_metadata(estimator, spark_model, params):
         if log_models:
             if _should_log_model(spark_model):
                 # TODO: support model signature
+                from mlflow.models.signature import infer_signature
+
+                def _cast_spark_df_with_vector_to_array(input_spark_df):
+                    """
+                    Finds columns of vector type in a spark dataframe and
+                    casts them to array<double> type.
+
+                    :param input_spark_df:
+                    :return: a spark dataframe with vector columns transformed to array<double> type
```

Comment:
@bali0019 @BenWilson2 Are columns of type `VectorUDT` the only non-scalar columns we have to special case? What about other `ArrayType` columns, e.g. string arrays?

Reply:
Other types (primitive type or complex type constructed from primitive type) can convert to pandas dataframe directly. I don't think they need special handling.

Reply:
Just to confirm, I am going to add a few more unit tests that will cover all DataTypes in spark to make sure auto-logging doesn't fail for some types.